### PR TITLE
Enhance the production example

### PR DESCRIPTION
**Pull request by marcofranssen**
To be double checked with latest release. https://github.com/spiffe/helm-charts/releases/tag/spire-0.5.0

### CODE REVIEW

File: examples/production/values.yaml [hunk 1]
```diff
@@ -37,7 +21,43 @@ spire-server:
       drop: [ALL]
     seccompProfile:
       type: RuntimeDefault
+
+  topologySpreadConstraints:
+    - maxSkew: 1
+      topologyKey: topology.kubernetes.io/zone
```
Comment (kfox1111) [inline examples/production/values.yaml:27]: I don't think a lot of clusters have zone defined, and DoNotSchedule together might break things?

File: examples/production/values.yaml [hunk 2]
```diff
@@ -37,7 +21,43 @@ spire-server:
...
+    enabled: true
+    size: 10Gi
+    accessMode: ReadWriteOnce
+    storageClass: gp3-encrypted
+
+  jwtIssuer: &jwtIssuer oidc-spire.eu-west-1-marco-test.my-domain.com
+  ca_subject:
+    country: "NL"
+    organization: "Marco Home Office"
+    common_name: spire.eu-west-1-marco-test.my-domain.com
```
Comment (kfox1111) [inline examples/production/values.yaml:52]: Should probably leave the jwtIssuer, ca_subject, clusterName and trustDomain alone?

File: examples/production/values.yaml [hunk 3]
```diff
@@ -47,10 +67,39 @@ spire-server:
...
+
+  resources:
+    requests:
+      cpu: 100m
+      memory: 32Mi
+    limits:
+      cpu: 200m
+      memory: 128Mi
+
+  nodeDriverRegistrar:
```
Comment (kfox1111) [inline examples/production/values.yaml:95]: This chunk should be removed. we dont have it.

File: examples/production/values.yaml [hunk 4]
```diff
@@ -60,3 +109,69 @@ spire-agent:
...
+  resources:
+    requests:
+      cpu: 100m
+      memory: 64Mi
+    limits:
+      cpu: 500m
+      memory: 512Mi
+
+  clusterName: *clusterName
+  trustDomain: *trustDomain
```
Comment (kfox1111) [inline examples/production/values.yaml:124]: Shouldn't have clustername/domain here.

File: examples/production/values.yaml [hunk 5]
```diff
@@ -60,3 +109,69 @@ spire-agent:
...
+
+  resources:
+    requests:
+      cpu: 100m
+      memory: 64Mi
+    limits:
+      cpu: 200m
+      memory: 256Mi
+
+  config:
```
Comment (kfox1111) [inline examples/production/values.yaml:168]: From here down should probably drop out too.

File: examples/production/values.yaml (hunk 3 above)
Comment (marcofranssen) [reply examples/production/values.yaml:95]: We do have that see the daemonset. There is a second container.

Comment (kfox1111) [reply examples/production/values.yaml:95]: Doh. I misread that and was thinking about the other k8s sync thing. Sorry.

File: examples/production/values.yaml [hunk 6]
```diff
@@ -41,7 +23,17 @@ spire-server:
       drop: [ALL]
     seccompProfile:
       type: RuntimeDefault
+
+  dataStorage:
+    enabled: true
+    size: 10Gi
```
Comment (kfox1111) [inline examples/production/values.yaml:29]: The db only seems to be used for a very small number of entries, even on a big cluster. 10Gi seems overkill.
1Gi? 
I'd recommend even smaller, but I've seen provisioning break on volumes < 1Gi.

File: examples/production/values.yaml [hunk 7]
```diff
@@ -41,7 +23,17 @@ spire-server:
...
+    accessMode: ReadWriteOnce
+
+  logLevel: info
+
   controllerManager:
+    enabled: true
```
Comment (kfox1111) [inline examples/production/values.yaml:35]: Its already defaulted to enabled.

**Review by kfox1111 — CHANGES_REQUESTED** [review]
Mostly looking good.

File: examples/production/values.yaml [hunk 8]
```diff
@@ -41,7 +23,17 @@ spire-server:
       drop: [ALL]
     seccompProfile:
       type: RuntimeDefault
+
+  dataStorage:
```
[the remark this answers is not in the document]
Comment (marcofranssen) [reply examples/production/values.yaml:27]: yup requires changes on the chart as well

**Review by kfox1111 — APPROVED** [review]
lgtm

**Pull request merged**